source/src/main/java/nl/aerius/fileserver/storage/FileController.java

```diff
@@ -76,6 +76,31 @@ public ResponseEntity<Void> putFile(final @PathVariable String uuid, final @Path
     return ResponseEntity.badRequest().build();
   }
 
+  /**
+   * Copy a file.
+   *
+   * @param sourceUuid uuid of file to copy from
+   * @param destinationUuid uuid of file to copy to
+   * @param filename filename of the file to put
+   * @param expires optional expires header value should conform to RFC 1123
+   */
+  @PutMapping(value = COPY_PATH)
+  public ResponseEntity<Void> copyFile(@PathVariable final String sourceUuid, final @PathVariable String destinationUuid,
+      final @PathVariable String filename, @RequestParam(name = "expires", required = false) final String expires) {
+    try {
+      FilenameUtil.validateParameters(destinationUuid, filename);
```

Comment:
Also validate the sourceUuid.

Reply:
Can do that I guess, but in this case the sourceUuid is used for retrieval. So it would be like validating the UUID in the GET method. And if it's unknown it'll cause an error anyway. So seems unnecessary?

Reply:
oh I see we validate it on GET as well. I'll add it.